cmd/sops/version.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/blang/semver"
+	"gopkg.in/urfave/cli.v1"
+)
+
+const version = "2.0.0"
+
+func printVersion(c *cli.Context) {
+	out := fmt.Sprintf("%s %s\n", c.App.Name, c.App.Version)
+	upstreamVersion, err := retrieveLatestVersionFromUpstream()
+	if err != nil {
+		out += fmt.Sprintf("[warning] failed to retrieve latest version from upstream: %v\n", err)
+	}
+	outdated, err := A_is_newer_than_B(upstreamVersion, version)
+	if err != nil {
+		out += fmt.Sprintf("[warning] failed to compare current version with latest: %v\n", err)
+	}
+	if outdated {
+		out += fmt.Sprintf("[info] sops %s is available, update with `go get -u go.mozilla.org/sops/cmd/sops`\n", upstreamVersion)
+	}
+	fmt.Fprintf(c.App.Writer, "%s", out)
+}
+
+// A_is_newer_than_B takes 2 semver strings are returns true
+// is the A is newer than B, false otherwise
+func A_is_newer_than_B(A, B string) (bool, error) {
+	vA, err := semver.Make(A)
+	if err != nil {
+		return false, err
+	}
+	vB, err := semver.Make(B)
+	if err != nil {
+		return false, err
+	}
+	if vA.Compare(vB) > 0 {
+		// vA is newer than vB
+		return true, nil
+	}
+	return false, nil
+}
+
+// retrieveLatestVersionFromUpstream gets the latest version from the source code at Github
+func retrieveLatestVersionFromUpstream() (string, error) {
+	resp, err := http.Get("https://raw.githubusercontent.com/mozilla/sops/master/cmd/sops/version.go")
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	scanner := bufio.NewScanner(resp.Body)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if len(line) > 20 && line[0:17] == `const version = "` {
```

Comment:
Using a non Go file is impractical because the compiled binary would not know its own version without reading that file, which may or may not exist on the file system.